### ISSUE TITLE
spiralMatrix.py

### ISSUE THREAD

**Issue by douglas-cerrato** [douglas-cerrato/LeetCode-75#12]

**Comment by douglas-cerrato** [comment]
### Spiral Matrix
-----------------------------------------
Given an ```m x n``` ```matrix```, return all elements of the ```matrix``` in spiral order.
# Example 1:
```
Input: matrix = [[1,2,3],[4,5,6],[7,8,9]]
Output: [1,2,3,6,9,8,7,4,5]
```
# Example 2:
```
Input: matrix = [[1,2,3,4],[5,6,7,8],[9,10,11,12]]
Output: [1,2,3,4,8,12,11,10,9,5,6,7]
```

**Comment by douglas-cerrato** [comment]
Here is our understanding of how we will change directions in the matrix to spiral so far. 

```python
    # If we were to view the matrix in a way where each value
    # is the index inside the matrix, it would like something 
    # like this:
    
    # 00  01  02                00  01  02  03
    # 10  11  12       or       10  11  12  13   
    # 20  21  22                20  21  22  23
    #                           30  31  32  33

    # Therefore, we can use this understanding to figure out how
    # we would go about navigating this matrix. This will help us
    # in terms of spiraling it. 

    # To iterate through the matrix by a direction. Here is how 
    # we would do it.

    # Assuming we view each index in this matrix as a (x,y) coordinate

    # Right: y + 1
    # Down: x + 1
    # Left: y - 1
    # Up: x - 1

    # This would mean we would have to add one position unit to either x or y
    # to view the current value at that coordinate. So now just need to figure out
    # how to spiral through the matrix (with hopes of doing with the fastest time
    # complexity)

```

This can be found in ee1b8630fd0a59dc7ba6ed47f8c517a608402b56

**Comment by douglas-cerrato** [comment]
Updating a mistake I made here. So techincally if we are viewing the matrix in a x,y graph the issue that arises is that python finds an index of a matrix in y,x format. So here is the updated movement rules:

```python
    # If we were to view the matrix in a way where each value
    # is the index inside the matrix, it would like something 
    # like this:
    
    # 00  01  02                00  01  02  03
    # 10  11  12       or       10  11  12  13   
    # 20  21  22                20  21  22  23
    #                           30  31  32  33

    # Therefore, we can use this understanding to figure out how
    # we would go about navigating this matrix. This will help us
    # in terms of spiraling it. 

    # To iterate through the matrix by a direction. Here is how 
    # we would do it.

    # Assuming we view each index in this matrix as a (x,y) coordinate

    # Right: x + 1
    # Down: y + 1
    # Left: x - 1
    # Up: y - 1

```

assuming we look at the variables like [y][x] as this is how python selects variables through index

**Comment by douglas-cerrato** [comment]
Figured out how to traverse the matrix. At first I made a bunch of if statements detecting where we are at in the code and traversing off of so while in a while loop. However my issue here was that I was deleting as I traversed, so modifying the matrix made it hard to traverse it correctly. I decided to take an approach of setting variables to the top, bottom, left and right of the matrix. Then I would do one loop through the matrix and update the boundaries as I go. With this, I can keep track of where I am at in the matrix at all times without modifying it
